File: Difficulty: Easy/Missing in Array/missing-in-array.java
//{ Driver Code Starts
import java.io.*;
import java.util.*;

class Geeks {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine());
        for (int g = 0; g < t; g++) {
            String[] str = (br.readLine()).trim().split(" ");
            int arr[] = new int[str.length];
            for (int i = 0; i < str.length; i++) {
                arr[i] = Integer.parseInt(str[i]);
            }
            System.out.println(new Solution().missingNumber(arr));
            System.out.println("~");
        }
    }
}
// } Driver Code Ends


// User function Template for Java
class Solution {
    int missingNumber(int arr[]) {
        // code here
         int n = arr.length + 1;
         int expectedSum = n * (n + 1) / 2;

        // Calculate the sum of elements in the given array
        int actualSum = 0;
        for (int num : arr) {
            actualSum += num;
        }

        // The missing element is the difference between the expected sum and the actual sum
        return expectedSum - actualSum;
    }
}

File: Difficulty: Easy/Missing in Array/missing-in-array.cpp
class Solution {
  public:
    int missingNum(vector<int>& arr) {
        // code here
        if(arr.size() == 1 && arr[0] == 1) return 2;
        else if(arr.size() == 1 && arr[0] == 2) return 1;
        
        sort(arr.begin(), arr.end());
        
        for(int i = 0; i <= arr.size(); i++)
        {
            if(arr[i] != i+1) return i+1;
        }
        
        return arr.size() + 1;
    }
};